### ISSUE TITLE
Runtime binding errors logged but not viewable

### ISSUE THREAD

**Issue by breyed** [xamarin/Xamarin.Forms#1517]
[BindingExpression.ApplyCore](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Core/BindingExpression.cs) calls `Log.Warning` if there's a runtime problem with XAML. However, these errors never show up in Visual Studio for Windows.

When debugging using Visual Studio for Windows, any binding errors should minimally show up in the Output Window in the Debug section. Even better would be an option to break execution when such a warning occurs.

The problem occurs with at least UWP and iOS.

**Comment by breyed** [comment]
As a side note, in my debugging, I never knew that there were unseen warnings until I encountered a different bug in which Protobuf throws an exception when it tries to access its [file descriptor](https://developers.google.com/protocol-buffers/docs/reference/csharp-generated#structure) on iOS if the Protobuf .cs file is in a .NET Standard library. The XAML logging code calls `ToString`, which triggers the iOS-netstandard bug and causes the iOS app to crash. The iOS-netstandard bug wouldn't matter if I could identify and fix whatever XAML problem is causing the warning in the first place.

**Comment by hartez** [comment]
For UWP, the error messages are showing up in the output window (Debug) in Visual Studio.
For iOS, however, they are definitely not appearing. 

It looks like the Forms.Init method for iOS is logging using `Trace.WriteLine`; I'm not sure where that ends up getting written for iOS.

**Comment by hartez** [comment]
@breyed As a workaround for the time being, you can add an additional log listener in the `AppDelegate` for your iOS application. Before calling `LoadApplication`, do something like this:
```
Log.Listeners.Add(new DelegateLogListener((c, m) => Debug.WriteLine(m, c)));
```

This will write the Log output to the debug window in VS.

**Comment by breyed** [comment]
Do you need to do anything to enable the runtime error output on UWP? I don't get any, even if it hack up the XAML to bind to nonexistent properties.

**Comment by hartez** [comment]
@breyed I stand corrected - I was testing on a debug version of Forms. In the released packages, the call to send the output to Debug isn't there. So you are correct - the log output is _not_ showing up in the VS debug output for a UWP project.

The same method I posted above for iOS will work for UWP. After you call `Xamarin.Forms.Forms.Init(e);` in your UWP project's `App.xaml.cs`, you can set up your own log listener:

`Log.Listeners.Add(new DelegateLogListener((c, m) => Debug.WriteLine(m, c)));`

And the output will show up in the debug console in VS.

**Comment by akamud** [comment]
I'm not sure I see the whole picture here, but isn't a fix for this just changing the `Trace.WriteLine` to `Debug.WriteLine` in all `Forms.cs`?

**Comment by hartez** [comment]
@akamud When we build the Release version of Forms, the `Debug.WriteLine` calls will be compiled out (because the `DEBUG` flag is not specified). 

The best option here is probably something like `Trace.WriteLine` in the Forms packages, and updating our templates to include default TraceListeners which are are active in Debug.

**Comment by jfversluis** [comment]
Isn't this fixed by #2911 and #3873 when it gets merged?
